service/collector/rate_limit.go

```diff
@@ -121,61 +120,25 @@ func NewRateLimit(config RateLimitConfig) (*RateLimit, error) {
 }
 
 func (u *RateLimit) Collect(ch chan<- prometheus.Metric) error {
-	// We need all CRs to gather all subscriptions below.
-	var crs []providerv1alpha1.AzureConfig
-	{
-		mark := ""
-		page := 0
-		for page == 0 || len(mark) > 0 {
-			opts := metav1.ListOptions{
-				Continue: mark,
-			}
-			list, err := u.g8sClient.ProviderV1alpha1().AzureConfigs(metav1.NamespaceAll).List(opts)
-			if err != nil {
-				return microerror.Mask(err)
-			}
-
-			crs = append(crs, list.Items...)
-
-			mark = list.Continue
-			page++
-		}
+	clientSets, err := credential.GetAzureClientSetsFromCredentialSecrets(u.k8sClient, u.environmentName)
```

Comment:
This collector will expose metrics not just by subscription. It will expose metric for every subscription id, client id and tenant id.